app/services/webhook/webhook.go

```diff
@@ -0,0 +1,173 @@
+	var err error
+
+	fullName := fmt.Sprintf("%d-%s", webhook.ID, webhook.Name)
+	result.Url, err = executeTemplate(fmt.Sprintf("%s-url", fullName), webhook.Url, props)
+	if err != nil {
+		return resultWithError(ctx, "Could not parse webhook URL template", err.Error(), result)
+	}
+	result.Content, err = executeTemplate(fmt.Sprintf("%s-content", fullName), webhook.Content, props)
+	if err != nil {
+		return resultWithError(ctx, "Could not parse webhook content template", err.Error(), result)
+	}
+
+	httpRequest := &cmd.HTTPRequest{
+		URL:       result.Url,
+		Body:      strings.NewReader(result.Content),
+		Method:    webhook.HttpMethod,
+		Headers:   webhook.AdditionalHttpHeaders,
+		BasicAuth: nil,
+	}
+	err = bus.Dispatch(ctx, httpRequest)
+	if err != nil {
+		return resultWithError(ctx, "Could not execute webhook HTTP request", err.Error(), result)
+	}
+	result.StatusCode = httpRequest.ResponseStatusCode
+	if result.StatusCode >= http.StatusBadRequest {
+		fullResponse := fmt.Sprintf("%d %s:\n%s", result.StatusCode, http.StatusText(result.StatusCode), httpRequest.ResponseBody)
+		return resultWithError(ctx, "Webhook HTTP request returned an error response code", fullResponse, result)
+	}
+
+	result.Success = true
+	log.Infof(ctx, "Webhook #@{ID:yellow} @{Name:blue} finished with @{Code:magenta}", dto.Props{
+		"ID":   webhook.ID,
+		"Name": webhook.Name,
+		"Code": result.StatusCode,
+	})
+	return result, nil
+}
+
+func previewWebhook(ctx context.Context, c *cmd.PreviewWebhook) error {
+	c.Result = &entity.WebhookPreviewResult{}
+	var err error
+	props := dummyTriggerProps(ctx, c.Type)
+
+	c.Result.Url.Value, err = executeTemplate("preview-url", c.Url, props)
+	if err != nil {
+		c.Result.Url.Message = "Could not parse webhook URL template"
+		c.Result.Url.Error = err.Error()
+		// Do not propagate error: it's a preview
+	}
+	c.Result.Content.Value, err = executeTemplate("preview-content", c.Content, props)
+	if err != nil {
+		c.Result.Content.Message = "Could not parse webhook content template"
+		c.Result.Content.Error = err.Error()
+		// Do not propagate error: it's a preview
+	}
+
+	return nil
+}
+
+func executeTemplate(name, text string, props dto.Props) (string, error) {
+	tmpl, err := tpl.GetTextTemplate(name, text)
+	if err != nil {
+		return "", err
+	}
+
+	replacedText, err := tpl.Execute(tmpl, props)
+	if err != nil {
+		return "", err
+	}
+
+	return replacedText, nil
+}
+
+func resultWithError(ctx context.Context, message, error string, result *entity.WebhookTriggerResult) (*entity.WebhookTriggerResult, error) {
+	result.Success = false
+	result.Message = message
+	result.Error = error
+
+	log.Warnf(ctx, "@{Message} (ID: @{ID:yellow}, Name: @{Name:blue}): @{Error:red}", dto.Props{
+		"Message": message,
+		"ID":      result.Webhook.ID,
+		"Name":    result.Webhook.Name,
+		"Error":   error,
+	})
+
+	webhooks := &query.MarkWebhookAsFailed{ID: result.Webhook.ID}
```

Comment:
after a webhook is marked as Failed we never process it again? unless someone re-enable it?

Reply:
Yes, it's the point. An extra task here would be to notify the admins by email, but I didn't find the time to do it, and with the internationalization going, emails were being converted to translated templates at the time and I didn't risked myself into this...
But in theory, when a Webhook has failed, there's a reason behind this and I found this solution instead of keeping failing logs in an extra DB table (but maybe it would be better to do this, and thus being able in the admin panel to see the real failed request instead of a manually triggered one. It would make sense, what are your thought about this?).

Reply:
Yeah, it's a lot more complex and we can implement that separately 👍

An email is probably a good starting point, otherwise webhook will get disabled silently. We can add this in a following PR